graylog2-server/src/test/java/org/graylog2/inputs/codecs/JsonPathCodecTest.java

```diff
@@ -41,39 +51,75 @@ public void testReadResultingInSingleInteger() throws Exception {
         String path = "$.state";
 
-        Map<String, Object> result = new JsonPathCodec(configOf(CK_PATH, path)).read(json);
-        assertEquals(1, result.size());
-        assertEquals("uploaded", result.get("result"));
+        Map<String, Object> result = new JsonPathCodec(configOf(CK_PATH, path), objectMapperProvider.get()).read(json);
+        assertThat(result.size()).isEqualTo(1);
+        assertThat(result.get("result")).isEqualTo("uploaded");
+    }
+
+    @Test
+    public void testReadResultingInSingleStringFullJson() throws Exception {
+        RawMessage json = new RawMessage("{\"url\":\"https://api.github.com/repos/Graylog2/graylog2-server/releases/assets/22660\",\"download_count\":76185,\"id\":22660,\"name\":\"graylog2-server-0.20.0-preview.1.tgz\",\"label\":\"graylog2-server-0.20.0-preview.1.tgz\",\"content_type\":\"application/octet-stream\",\"state\":\"uploaded\",\"size\":38179285,\"updated_at\":\"2013-09-30T20:05:46Z\"}".getBytes(StandardCharsets.UTF_8));
+        String path = "$.state";
+
+        Message result = new JsonPathCodec(configOf(CK_PATH, path, CK_FLATTEN, true), objectMapperProvider.get()).decode(json);
+        assertThat(result.getField("state")).isEqualTo("\"uploaded\"");
     }
 
     @Test
     public void testReadFromMap() throws Exception {
         String json = "{\"store\":{\"book\":[{\"category\":\"reference\",\"author\":\"Nigel Rees\",\"title\":\"Sayings of the Century\",\"price\":8.95},{\"category\":\"fiction\",\"author\":\"Evelyn Waugh\",\"title\":\"Sword of Honour\",\"price\":12.99,\"isbn\":\"0-553-21311-3\"}],\"bicycle\":{\"color\":\"red\",\"price\":19.95}}}";
         String path = "$.store.book[?(@.category == 'fiction')].author";
 
-        Map<String, Object> result = new JsonPathCodec(configOf(CK_PATH, path)).read(json);
-        assertEquals(1, result.size());
-        assertEquals("Evelyn Waugh", result.get("result"));
+        Map<String, Object> result = new JsonPathCodec(configOf(CK_PATH, path), objectMapperProvider.get()).read(json);
+        assertThat(result.size()).isEqualTo(1);
+        assertThat(result.get("result")).isEqualTo("Evelyn Waugh");
+    }
+
+    @Test
+    public void testReadFromMapFullJson() throws Exception {
+        RawMessage json = new RawMessage("{\"store\":{\"book\":[{\"category\":\"reference\",\"author\":\"Nigel Rees\",\"title\":\"Sayings of the Century\",\"price\":8.95},{\"category\":\"fiction\",\"author\":\"Evelyn Waugh\",\"title\":\"Sword of Honour\",\"price\":12.99,\"isbn\":\"0-553-21311-3\"}],\"bicycle\":{\"color\":\"red\",\"price\":19.95}}}".getBytes(StandardCharsets.UTF_8));
+        String path = "$.store.book[?(@.category == 'fiction')].author";
+
+        Message result = new JsonPathCodec(configOf(CK_PATH, path, CK_FLATTEN, true), objectMapperProvider.get()).decode(json);
+        assertThat("\"Evelyn Waugh\"").isEqualTo(result.getField("store.book1.author"));
```

Comment:
I think here should be switched the assertThat and isEqualTo arguments. Left the actual value, right the expected.